### ISSUE TITLE
LoggingSystemProperties writes unwanted logging during reinit when logging properties are used

### ISSUE THREAD

**Issue by rfscholte** [spring-projects/spring-boot#37498]
The bug was discovered when working with spring-cloud-stream-test-support [org.springframework.cloud.stream.binder.BinderTestEnvironmentPostProcessor](https://github.com/spring-cloud/spring-cloud-stream/blob/main/core/spring-cloud-stream-test-support/src/main/java/org/springframework/cloud/stream/binder/BinderTestEnvironmentPostProcessor.java#L39-L40) and transformed in [a minimal testcase](https://github.com/rfscholte/sb3_LoggingSystemProperties_bug).

When running this project it will succeed, however it shows logging that contains a warning. My goal it to have a warningless build (and even tests without any logging), but this one cannot be removed. It is also not clear what the issue is and how to solve it.

```
[INFO] Running com.sourcegrounds.lesslogging.ApplicationIT
19:56:25,401 |-WARN in Logger[org.springframework.core.env.PropertySourcesPropertyResolver] - No appenders present in context [default] for logger [org.springframework.core.env.PropertySourcesPropertyResolver].
19:56:25,424 |-INFO in ch.qos.logback.classic.model.processor.RootLoggerModelHandler - Setting level of ROOT logger to OFF
19:56:25,425 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@680362a - Propagating OFF level on Logger[ROOT] onto the JUL framework
19:56:25,427 |-INFO in ch.qos.logback.core.model.processor.DefaultProcessor@3569edd5 - End of configuration.
19:56:25,427 |-INFO in org.springframework.boot.logging.logback.SpringBootJoranConfigurator@1f651cd8 - Registering current configuration as safe fallback point

[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.84 s - in com.sourcegrounds.lesslogging.ApplicationIT
```

It seems the root cause is in the LoggingSystemProperties in case one of these logging-properties is set, because the PropertyResolver wants to log it when a key is found.

**Comment by wilkinsona** [comment]
This looks very similar to #34505. We'll have to investigate further to determine if it's a duplicate of just a similar problem with a different root cause.

**Comment by rfscholte** [comment]
Based on the title it looks a different usecase with similar symptoms, but responses further in the thread point to the same rootcause.
Maybe worth mentioning: this info might be interesting when debugging the logging system, just like the information that is cleared in https://github.com/spring-projects/spring-boot/blob/7de770f6a191d16e3c120671de8b29349d95d57d/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/logback/LogbackLoggingSystem.java#L326

**Comment by wilkinsona** [comment]
Thanks for the minimal reproducer. I have a fix for this that also fixes #34505 so I'll close this one as a duplicate of the earlier issue.
